### PR TITLE
[En-89] Displayed only Active contracts in Contracts entry

### PR DESCRIPTION
**Pull request by baremarco**
# Description :pencil:

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

_Displayed only Active contracts in Contracts entry_

- create FilteredDatagrid for render content based on a specific condition
- migrate ChustomizableChipField component to tsx
- fix warning about sssign object to a variable before exporting as module default 

## Link to ticket :link:

https://entropyteam.atlassian.net/browse/EN-89

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? :microscope:

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- tested contract grid full behaviour, bulk actions as delete and individual actions as edit

### CODE REVIEW

File: src/components/fields/CustomizableChipField.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { useRecordContext } from "react-admin";
+import Chip from "@mui/material/Chip";
+import { IPaymentSettlement } from "../../types";
+import { SxProps } from "@mui/material";
+
+interface CustomizableChipFieldProps {
+  children: (record: IPaymentSettlement) => string | null;
```
Comment (agustinb4) [inline src/components/fields/CustomizableChipField.tsx:8]: Could we decouple this component from the IPaymentSettlement? maybe passing it as param?

File: src/components/fields/CustomizableChipField.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    | "error"
+    | "info"
+    | "success"
+    | "warning";
+  size?: "small" | "medium";
+  sx?: SxProps;
+}
+
+export const CustomizableChipField = (props: CustomizableChipFieldProps) => {
+  const record = useRecordContext<IPaymentSettlement>(props);
```
Comment (agustinb4) [inline src/components/fields/CustomizableChipField.tsx:23]: Same here

**Review by agustinb4 — COMMENTED** [review]
Looks good! Some comments

**Review by FacuOrellana — APPROVED** [review]
🙌🏻

**Review by agustinb4 — APPROVED** [review]

**Pull request merged**